pkg/controller/storagecluster/noobaa_system_reconciler_test.go

```diff
@@ -121,6 +121,100 @@ func TestEnsureNooBaaSystem(t *testing.T) {
 	}
 }
 
+func TestNooBaaReconcileStratgy(t *testing.T) {
+	namespacedName := types.NamespacedName{
+		Name:      "noobaa",
+		Namespace: "test_ns",
+	}
+
+	cephCluster := cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephClusterFromString(namespacedName.Name),
+			Namespace: namespacedName.Namespace,
+		},
+	}
+	cephCluster.Status.State = cephv1.ClusterStateCreated
+
+	cases := []struct {
+		label          string
+		namespacedName types.NamespacedName
+		sc             v1.StorageCluster
+		noobaa         v1alpha1.NooBaa
+		isCreate       bool
+	}{
+		{
+			label: "case 1", //ensure default create logic
+			sc: v1.StorageCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+			},
+			isCreate: true,
+		},
+		{
+			label: "case 2", //ensure create logic
+			sc: v1.StorageCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+				Spec: v1.StorageClusterSpec{
+					MultiCloudGateway: &v1.MultiCloudGatewaySpec{
+						ReconcileStrategy: "manage",
+					},
+				},
+			},
+			isCreate: true,
+		},
+		{
+			label: "case 3", //ensure unknown value logic
+			sc: v1.StorageCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+				Spec: v1.StorageClusterSpec{
+					MultiCloudGateway: &v1.MultiCloudGatewaySpec{
+						ReconcileStrategy: "foo",
+					},
+				},
+			},
+			isCreate: true,
+		},
+		{
+			label: "case 4", //ensure ignore logic
+			sc: v1.StorageCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+				Spec: v1.StorageClusterSpec{
+					MultiCloudGateway: &v1.MultiCloudGatewaySpec{
+						ReconcileStrategy: "ignore",
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		reconciler := getReconciler(t, &v1alpha1.NooBaa{})
+		reconciler.client.Create(context.TODO(), &cephCluster)
+
+		err := reconciler.ensureNoobaaSystem(&c.sc, noobaaReconcileTestLogger)
+		assert.NoError(t, err)
+
+		if c.isCreate {
+			err = reconciler.client.Get(context.TODO(), namespacedName, &v1alpha1.NooBaa{})
```

Comment:
Can we put this above the `if c.isCreate` check? (Reducing the blocks to the assert statement)

Reply:
Sure.